### PR TITLE
test: add frontend unit tests for utilities, components, and QueryEditor

### PR DESCRIPTION
**Pull request by diericd**
- Add tests for src/util/util.ts (isRegex, isValidRegex, isUUID)
- Add tests for src/util/semver.ts (semverCompare, isFeatureEnabled)
- Add tests for src/QueryEditor/util.ts (selectable, aggregation filters, asset matching, tag converters, debouncePromise, migration helpers)
- Add tests for src/QueryEditor/RawQueryEditor.tsx (loading state, database fetch, CodeEditor visibility)
- Add tests for src/components/Cascader/Cascader.tsx (flattenOptions, getSearchableOptions memoization, setInitialValue)
- Add tests for src/QueryEditor.tsx (tab rendering, sub-editor switching, onChange propagation)
- Fix jest.config.js to pin SWC target to es2022 (installed @swc/core does not support es2023)
- Add TextEncoder/TextDecoder polyfill to jest-setup.js for @grafana/ui compatibility with jsdom

### CODE REVIEW

File: jest-setup.js [hunk 1]
```diff
@@ -1,2 +1,6 @@
 // Jest setup provided by Grafana scaffolding
 import './.config/jest-setup'
+
+// Polyfill TextEncoder/TextDecoder for @grafana/ui compatibility with jsdom
+import { TextDecoder, TextEncoder } from 'util'
+Object.assign(global, { TextDecoder, TextEncoder })
```
[the remark this answers is not in the document]
Comment (diericd) [reply jest-setup.js:6]: This is standard Jest test setup — the polyfill only runs within the Jest process and is scoped to the test environment via `setupFilesAfterSetup`. No production impact.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -37,7 +37,32 @@ jobs:
       - name: Lint
         run: pnpm run lint
       - name: Unit tests
-        run: pnpm run test:ci
+        run: pnpm run test:coverage
+
+      - name: Extract coverage percentage
+        id: coverage
+        run: |
+          PCT=$(jq -r '.total.lines.pct' coverage/coverage-summary.json)
```
[the remark this answers is not in the document]
Comment (diericd) [reply .github/workflows/ci.yml]: Fixed — added a file existence guard that defaults to 0% if `coverage-summary.json` is missing.

File: src/util/util.test.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+  it('returns false for an empty string', () => {
+    expect(isRegex('')).toBe(false)
+  })
+
+  it('returns false for a single slash', () => {
+    expect(isRegex('/')).toBe(false)
+  })
+
+  it('returns false for two slashes (edge case: empty regex)', () => {
```
[the remark this answers is not in the document]
Comment (diericd) [reply src/util/util.test.ts]: Fixed — updated the test description to "returns true".

File: src/QueryEditor/util.test.ts [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    const query: MeasurementQuery = {
+      IsRegex: false,
+      Measurements: ['/sensor-.*/'],
+      Options: {} as any,
+    }
+    const result = migrateMeasurementQuery(query)
+    // The source sets IsRegex on the original query object (mutation), returns a shallow clone
+    expect(query.IsRegex).toBe(true)
+    expect(query.Regex).toBe('sensor-.*')
+    expect(query.Measurements).toEqual([])
```
[the remark this answers is not in the document]
Comment (diericd) [reply src/QueryEditor/util.test.ts]: Good catch — this was a real bug. Fixed `migrateMeasurementQuery` to mutate the clone (`measurementQuery`) instead of the input (`query`). Updated the test to assert on the return value and added a separate test verifying the original query is not mutated.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,7 @@
 # <img src="https://raw.githubusercontent.com/factrylabs/factry-historian-datasource/main/src/img/logo.svg" alt="Factry Historian Logo" height="30"> Factry Historian Datasource
 
+[![Coverage](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/factrylabs/REPLACE_WITH_GIST_ID/raw/factry-historian-coverage.json)](https://github.com/factrylabs/factry-historian-datasource/actions)
+
```
[the remark this answers is not in the document]
Comment (diericd) [reply README.md]: Removed the placeholder badge for now. Will add it back once the gist is properly configured.

**Review by vereecw — APPROVED** [review]
